packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -39,42 +40,63 @@ export class MongoLogManager {
     if (options.prefix) {
       if (!/^[a-z0-9_]+$/i.test(options.prefix)) {
         throw new Error(
-          'Prefix must only contain letters, numbers, and underscores'
+          'Prefix must only contain letters, numbers, and underscores',
         );
       }
     }
     this._options = options;
   }
 
-  private async deleteFile(path: string): Promise<void> {
+  private async deleteFile(filePath: string): Promise<void> {
     try {
-      await fs.unlink(path);
+      await fs.unlink(filePath);
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
     } catch (err: any) {
       if (err?.code !== 'ENOENT') {
-        this._options.onerror(err as Error, path);
+        this._options.onerror(err as Error, filePath);
       }
     }
+
+    let basePath = path.dirname(filePath);
+    while (basePath !== this._options.directory) {
+      // Delete empty directories up to the base directory
+      try {
+        await fs.rmdir(path.dirname(filePath));
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (err: any) {
+        // If the directory is not empty, we stop deleting.
+        if (err?.code === 'ENOTEMPTY' || err?.code === 'ENOENT') {
```

Comment:
These would be the most common errors that `rmdir()` could encounter, but I don't think it makes sense to continue in _any_ error situation here, right?